File: packages/nexus-sdk-js/src/analytics.ts
/**
 * NEXUS SDK Analytics Operations
 *
 * Provides access to API usage analytics for B2B clients.
 */

import type { UsageAnalytics, AnalyticsPeriod } from './types.js';

type RequestFn = <T>(method: string, path: string, body?: unknown) => Promise<T>;

/**
 * Analytics resource operations
 *
 * @example
 * ```typescript
 * // Get usage analytics for the last 30 days
 * const usage = await client.analytics.getUsage('30d');
 * console.log(`Total escrows: ${usage.total_escrows}`);
 * console.log(`Volume: ${usage.total_volume_atomic} piconero`);
 * ```
 */
export class AnalyticsResource {
  private readonly request: RequestFn;

  constructor(request: RequestFn) {
    this.request = request;
  }

  /**
   * Get usage analytics for the authenticated client
   *
   * GET /api/v1/analytics/usage
   *
   * @param period - Time period: '24h', '7d', '30d', or 'all' (default: '30d')
   */
  async getUsage(period?: AnalyticsPeriod): Promise<UsageAnalytics> {
    const params = new URLSearchParams();
    if (period) {
      params.set('period', period);
    }
    const query = params.toString();
    const path = query ? `/v1/analytics/usage?${query}` : '/v1/analytics/usage';
    return this.request<UsageAnalytics>('GET', path);
  }
}

File: packages/onyx-sdk-python/onyx_escrow/analytics.py
"""Analytics operations for Onyx SDK.

Provides API usage analytics and statistics.
"""

from __future__ import annotations

from typing import TYPE_CHECKING

from .types import UsageAnalytics

if TYPE_CHECKING:
    from .client import OnyxClient


class AnalyticsManager:
    """Manages analytics and usage reporting.

    Accessed via ``client.analytics``.

    Example::

        async with OnyxClient(api_key="nxs_...") as client:
            stats = await client.analytics.usage(period="30d")
            print(f"Total escrows: {stats.total_escrows}")
    """

    def __init__(self, client: OnyxClient) -> None:
        self._client = client

    async def usage(self, *, period: str = "30d") -> UsageAnalytics:
        """Get API usage analytics.

        Args:
            period: Time period filter ("24h", "7d", "30d", "all").

        Returns:
            Usage analytics for the specified period.
        """
        response = await self._client._request(
            "GET",
            "/api/v1/analytics/usage",
            params={"period": period},
        )
        return UsageAnalytics.model_validate(response)
